bin/propolis-server/src/lib/server.rs

```diff
@@ -337,6 +337,82 @@ async fn register_oximeter(
     Ok(registry)
 }
 
+/// Wrapper around a [`NexusClient`] object, which allows deferring
+/// the DNS lookup until accessed.
+///
+/// Without the assistance of OS-level DNS lookups, the [`NexusClient`]
+/// interface requires knowledge of the target service IP address.
+/// For some services, like Nexus, this can be painful, as the IP address
+/// may not have even been allocated when the Sled Agent starts.
+///
+/// This structure allows clients to access the client on-demand, performing
+/// the DNS lookup only once it is actually needed.
+struct Inner {
```

Comment:
naming nit: this is a pretty wide-ranging module with lots of distinct structs, so I'd go with `LazyNexusClientInner` for this.

Even better, I think, would be to move this from `omicron-nexus` to `nexus-client` or `omicron-common` so we can just reuse that code, but that's not needed for this change. (It might be worth filing an Omicron issue for, though.)

Reply:
I have done the re-name.

For the larger issue of moving this, you mean move the methods/functions here into Omicron proper?  Or put them in something local to this repo but in a common area?

Reply:
The former. My last comment was wrong. There's an identical-looking `LazyNexusClient` in the sled agent (`omicron/sled-agent/src/nexus.rs`; I misread the path and thought it was in `omicron-nexus` but that's in `omicron-sled-agent`). It'd be nice to consolidate these. Practically speaking, I'd do that by moving sled agent's lazy client into a smaller common crate that sled agent and Propolis (and anyone else who wants it) can link to, and I suspect that common crate should live in the Omicron repo. Any details beyond that are TBD :)